align/schema/constraint.py

```diff
@@ -659,30 +663,23 @@ class DoNotIdentify(SoftConstraint):
 class SymmetricBlocks(SoftConstraint):
     pairs: List[List[str]]
     direction: Literal['H', 'V']
+    mirror: Optional[bool] = True
 
     @types.validator('pairs', allow_reuse=True)
     def pairs_validator(cls, value):
-        '''
-        X = Align(2, 3, 'h_center')
-        Y = Align(4, 5, 'h_center')
-        Align(1, X, Y, 6, 'center')
-
-        '''
-        instances = get_instances_from_hacked_dataclasses(cls._validator_ctx())
+        # instances = get_instances_from_hacked_dataclasses(cls._validator_ctx())
         for pair in value:
             assert len(pair) >= 1, 'Must contain at least one instance'
             assert len(pair) <= 2, 'Must contain at most two instances'
             validate_instances(cls, pair)
         if not hasattr(cls._validator_ctx().parent.parent, 'elements'):
             # PnR stage VerilogJsonModule
             return value
-        if len(cls._validator_ctx().parent.parent.elements)==0:
-            #skips the check while reading user constraints
-            return value
         group_block_instances = [const.name for const in cls._validator_ctx().parent if isinstance(const, GroupBlocks)]
         for pair in value:
             # logger.debug(f"pairs {self.pairs} {self.parent.parent.get_element(pair[0])}")
             if len([ele for ele in pair if ele in group_block_instances])>0:
+                # TODO: Why is this skip still needed?
                 #Skip check for group block elements as they are added later in the flow
```

Comment:
This comment doesn't entirely make sense as the constraints should be valid throughout the flow. I feel like there is some missing hierarchical check over here.  Do we need to look within groupblocks and make sure the instances have the same parameters or something? Otherwise the check is incomplete and only works in certain situations.

Reply:
Yes, that's right. I can do the hierarchical parameter check for groupblock.

Reply:
fixed